plugins/modules/net_tools/nmcli.py

```diff
@@ -1695,6 +1695,11 @@ def _compare_conn_params(self, conn_info, options):
                     # Depending on version nmcli adds double-qoutes to gsm.apn
                     # Need to strip them in order to compare both
                     current_value = current_value.strip('"')
+                if key == self.mtu_setting and self.mtu is None:
+                    self.mtu = 0
+            elif key == self.mtu_setting:
+                # if self.mtu_setting parameter doesn't exist, NetworkManager behave like it's set to 'auto'
+                current_value = 'auto'
```

Comment:
```suggestion
```
I don't think this part is useful and can be removed. This just makes the diff show `"auto"` instead of `null` after supplying a value for `mtu` with dummy connections when it was not previously applied. Which is confusing since the value was in fact `null`.